### PR TITLE
Harden musefs-format bounds: OggArtSlice validation (#273) + checked aggregate arithmetic (#274)

### LINKED ISSUE #273: Validate OggArtSlice source bounds in RegionLayout
Adversarial audit finding from Task 6: `RegionLayout::validate` does not check the semantic bounds of `Segment::OggArtSlice`.

## What I found

`musefs-format/src/layout.rs` validates empty non-backing segments, `BackingAudio` range overflow, and total synthesized length overflow, but it does not verify that an Ogg art slice names a valid output window for its source art.

The missing invariants are:

- raw art slices: `offset + len <= art_total`
- base64 art slices: `offset + len <= b64_len(art_total)`
- both should use checked arithmetic

The reader path in `musefs-core/src/reader.rs` assumes these relationships when it computes `b64_window(*offset + within, n as u64, *art_total)` or when it streams raw art at `*offset + within`.

## Why this matters

Current Ogg producers appear to construct slices from concrete `PayloadChunk::Art.out` buffers, so the normal path should produce coherent segments. But `RegionLayout::validated` is the central validation boundary for layouts; fuzzing and future format producers should not need to independently rediscover the `OggArtSlice` constraints.

An invalid layout can currently pass validation and reach serving code, where it can produce unexpected read errors, short reads, or panics depending on which path observes the out-of-range slice.

## Suggested fix

Extend `RegionLayout::validate` to reject invalid `OggArtSlice` segments:

- compute the permitted output length as `art_total` for raw slices and `musefs_format::ogg::b64_len(art_total)` for base64 slices;
- reject checked-add overflow for `offset + len`;
- reject any slice whose end exceeds that permitted output length;
- add layout tests for raw overflow, base64 overflow, and `offset + len` overflow.

### LINKED ISSUE #274: Use checked aggregate length arithmetic in format synthesis builders
Adversarial audit finding from Task 8: several format synthesis builders rely on unchecked aggregate length arithmetic before the final format-size or `RegionLayout` validation boundary.

## What I found

The code has good per-field size guards, but some aggregate calculations still use plain `+`, `+=`, or `sum::<u64>()` over attacker-controlled DB-derived lengths.

Examples:

- `musefs-format/src/mp3.rs`: `frames_len += 10 + ...` and `10 + frames_len` in `build_id3v2_segments`.
- `musefs-format/src/mp4.rs`: `streamed_total += ...`, `covr_size = 8 + sum(16 + art_len)`, `ilst_size/meta_size/udta_size`, `new_moov_size`, and `new_mdat_payload_pos`.
- `musefs-format/src/wav.rs`: `riff_size = u32::try_from(body_len + 4)`.
- `musefs-format/src/flac.rs` and `musefs-format/src/ogg/mod.rs`: picture/body/comment aggregate lengths such as `framing.len() as u64 + art_len`, `prefix.len() as u64 + art_len`, and Vorbis/MBP value-length sums should be checked consistently.

In ordinary scanner-produced data these values are small. Under a hostile or externally written DB, however, `art.byte_len`, binary tag lengths, structural block bodies, and text values can be adversarial.

## Why this matters

A later `RegionLayout::validate` often catches total-length overflow before a corrupt layout is served, but that is too late and too indirect:

- debug builds can panic on arithmetic overflow instead of returning `FormatError::TooLarge`;
- release builds may wrap intermediate box/frame sizes and only fail later by accident;
- future changes could move or weaken the later validation and turn the wrapped size into emitted metadata.

This should fail closed at the exact format arithmetic boundary.

## Suggested fix

Replace aggregate size math in synthesis builders with checked helpers, for example:

- `checked_add` / `checked_sum` returning `FormatError::TooLarge`;
- format-specific helpers for `u32` box/frame/body sizes;
- tests that construct synthetic `ArtInput`/`BinaryTagInput` values near `u64::MAX` or multiple `i64::MAX`-sized lengths and assert `Err(FormatError::TooLarge)` rather than panic.

Keep `RegionLayout::validate` as the final belt-and-suspenders check, but do not depend on it to discover format arithmetic overflow.

### PR DESCRIPTION
**Pull request by Sohex**
Closes #273. Closes #274.

Makes `musefs-format` fail closed on adversarial, DB-derived lengths at the exact arithmetic/validation boundary instead of relying on later checks (or wrapping in release / panicking in debug).

## #273 — OggArtSlice source bounds in `RegionLayout::validate`
`validate` previously checked `OggArtSlice` length/total only structurally, never that the named output window lies within its source art. Added two `LayoutError` variants and a validation arm:
- raw slices: `offset + len <= art_total`
- base64 slices: `offset + len <= b64_len(art_total)`
- checked arithmetic throughout: `offset + len` overflow and `b64_len(art_total)` overflow both reject (`OggArtSliceRangeOverflow`); an in-range-but-past-source window rejects (`OggArtSliceOutOfBounds`).

New `b64_len_checked(img_total) -> Option<u64>`; the panicking `b64_len` now delegates to it (no behaviour change for real images). A boundary test pins that `b64_window`'s internal arithmetic stays overflow-free for any layout that passes `validate`.

## #274 — checked aggregate length arithmetic in the synthesis builders
New `pub(crate)` `size` module (`checked_add`/`checked_sum`, both mapping `u64` overflow to `FormatError::TooLarge`). Replaced unchecked `+`/`+=`/`sum::<u64>()` over attacker-controlled lengths in:
- **mp3** `build_id3v2_segments` — all frame-length accumulators (text/TXXX/COMM/USLT/POPM/UFID/binary-tag/art) and the returned total
- **mp4** `build_udta` / `synthesize` — covr/data/ilst/meta/udta box sizes, `new_moov_size`, `new_mdat_payload_pos`, **and** `freeform_binary_prefix`'s binary-tag box size (the one path with no upstream u32 bound on the DB length)
- **wav** `synthesize_layout` — RIFF size aggregate
- **flac** `synthesize_layout` — picture-block body aggregate
- **ogg** `build_packets_with_art` / `comment_packet_chunks` — VorbisComment picture `value_len` (now via `b64_len_checked`)

The format builders now fail closed with `TooLarge`; `RegionLayout::validate` remains the final belt-and-suspenders check.

## Notes
- Spec + plan included under `docs/superpowers/`.
- The mp4 `freeform_binary_prefix` site was caught in post-implementation review (the only binary-tag length not capped by a u32 field upstream); the analogous ogg-flac art path was found already guarded by `picture_prefix`'s u32 data-length field, so it was left unchanged.
- Each commit is independently green (full workspace suite via the pre-commit hook). The out-of-workspace `fuzz/` crate builds.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Hardened overflow-safe size accounting across MP3, MP4, WAV, FLAC, and OGG (new checked size helpers)
  * Added strict Ogg art-slice bounds validation and safer base64-length checks to prevent out-of-bounds/overflow
  * Oversized inputs now return clear errors instead of producing corrupted layouts

* **Documentation**
  * Added design and plan documents detailing the layout bounds hardening

* **Tests**
  * Added overflow-boundary and validation tests covering new checks and error paths

### CODE REVIEW

**Comment by Sohex** [comment]
@coderabbitai Re: the review finding on `mp4.rs` `ilst_inline_len`/`udta_total` (and `meta_inline_len`/`udta_inline_len`) — I looked into this and believe it's a false positive, so I've left those sites unchanged. Reasoning:

- `ilst_inline_len` sums **only `Segment::Inline(b)` real buffer lengths** (`b.len() as u64`), explicitly mapping streamed segments to `0`. Those buffers are materialized in memory, so their total can't approach `u64::MAX` without OOMing first (~18 EB). `meta_inline_len` / `udta_inline_len` only combine those buffer lengths with small constants, so the raw `+` there can't overflow either.
- The attacker-controlled, *unallocated* DB lengths (art/binary-tag payloads, which stream from the store rather than being materialized) flow into `streamed_total`, which **is** routed through `size::checked_add`.
- `udta_total` (the post-`build_udta` `.sum()` over segment lengths) only runs after `build_udta` succeeded, where `udta_size = checked_sum([8, udta_inline_len, streamed_total])?` followed by `u32::try_from(udta_size)` already bounds it; `udta_total <= udta_size <= u32::MAX`, so that sum can't overflow.

The guiding rule applied across the PR is: **check operands that are unallocated DB lengths, not real in-memory buffer lengths.** Adding `checked_sum` over `b.len()` would guard a physically impossible condition and obscure that invariant.

For the record, the other two findings were valid and are fixed: the OggFLAC picture body length (`ogg/mod.rs`, commit `99c2edc`) and the `freeform_binary_prefix` outer `8 + inner_len` box-size add (`mp4.rs`, commit `183305d`, with a regression test pinning the exact `inner_len == u64::MAX` window).

File: musefs-format/src/mp4.rs [hunk 1]
```diff
@@ -616,8 +617,8 @@ fn freeform_binary_prefix(mean: &str, name: &str, payload_len: u64) -> Result<Ve
     name_body.extend_from_slice(name.as_bytes());
     let name_box = boxed(b"name", &name_body)?;
 
-    let data_size = 8 + 8 + payload_len; // data header + type + locale + payload
-    let inner_len = mean_box.len() as u64 + name_box.len() as u64 + data_size;
+    let data_size = size::checked_add(16, payload_len)?; // data header + type + locale + payload
+    let inner_len = size::checked_sum([mean_box.len() as u64, name_box.len() as u64, data_size])?;
 
     let mut out = u32::try_from(8 + inner_len)
         .map_err(|_| FormatError::TooLarge)?
```
[the remark this answers is not in the document]
Comment (Sohex) [reply musefs-format/src/mp4.rs:625]: Fixed in 183305d — `let outer_len = size::checked_add(8, inner_len)?;`. Added a regression test (`freeform_binary_prefix_checked_outer_box_size_rejects_overflow`) using `payload_len = u64::MAX - 42`, which drives `inner_len` to exactly `u64::MAX` and so reaches this outer add — the earlier `u64::MAX` test tripped the `data_size` check first and never got here.
